orttraining/orttraining/core/optimizer/graph_transformer_utils.cc

```diff
@@ -104,6 +105,10 @@ std::vector<std::unique_ptr<GraphTransformer>> GeneratePreTrainingTransformers(
             horizontal_parallel_size, compatible_eps));
       }
       transformers.emplace_back(onnxruntime::make_unique<ComputationReductionTransformer>(compatible_eps));
+
+      if (config.transformer_layer_checkpoint) {
```

Comment:
>transformer_layer_checkpoint [](start = 17, length = 28)

Would it better to name it transformer_layer_recompute? checkpoint might cause confusing as it has different meanings in training state save/restore.